WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizerTests.swift

```diff
@@ -142,6 +221,25 @@ final class RemoteOrderSynchronizerTests: XCTestCase {
         XCTAssertEqual(item.quantity, input2.quantity)
     }
 
+    func test_sending_update_productID_input_updates_local_order() throws {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let synchronizer = RemoteOrderSynchronizer(siteID: sampleSiteID, flow: .creation, stores: stores)
+
+        // When
+        let input = OrderSyncProductInput(id: sampleInputID, product: .productID(sampleProductID), quantity: 1)
+        let input2 = OrderSyncProductInput(id: sampleInputID, product: .productID(sampleProductID), quantity: 2)
+        synchronizer.setProduct.send(input)
+        synchronizer.setProduct.send(input2)
+
+        // Then
+        let item = try XCTUnwrap(synchronizer.order.items.first)
+        XCTAssertEqual(item.itemID, input2.id)
+        XCTAssertEqual(item.productID, sampleProductID)
+        XCTAssertEqual(item.variationID, 0)
+        XCTAssertEqual(item.quantity, input2.quantity)
```

Comment:
```suggestion

        let initialProductState = OrderSyncProductInput(id: sampleInputID, product: .productID(sampleProductID), quantity: 1)
        synchronizer.setProduct.send(initialProductState)

        // When
        let productUpdateInput = OrderSyncProductInput(id: sampleInputID, product: .productID(sampleProductID), quantity: 2)
        synchronizer.setProduct.send(productUpdateInput)

        // Then
        let item = try XCTUnwrap(synchronizer.order.items.first)
        XCTAssertEqual(item.itemID, productUpdateInput.id)
        XCTAssertEqual(item.productID, sampleProductID)
        XCTAssertEqual(item.variationID, 0)
        XCTAssertEqual(item.quantity, productUpdateInput.quantity)
```

The initial state is part of the `//Given` section, then your test is for `//When` you update it

Reply:
Ah yes, I see what you mean. I changes this one and a couple of similar ones on https://github.com/woocommerce/woocommerce-ios/pull/9273/commits/78a0733acc832d1dc05625e572b9739d9d60aa2e